module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/MakeLinkTest.php

```diff
@@ -0,0 +1,219 @@
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class MakeLinkTest extends \PHPUnit\Framework\TestCase
+{
+    protected function getHelper()
+    {
+        $escapeHtml = new \Laminas\View\Helper\EscapeHtml();
+        $view = $this->createMock(\Laminas\View\Renderer\PhpRenderer::class);
+        $view->method('plugin')->will($this->returnValue($escapeHtml));
+
+        $helper = new MakeLink();
+        $helper->setView($view);
+        return $helper;
+    }
+
+    /**
+     * Test that responds to common inputs
+     *
+     * @return void
+     */
+    public function testLink()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals(
+            '<a href="https://vufind.org">text</a>',
+            $helper('text', 'https://vufind.org')
+        );
+
+        $this->assertEquals(
+            '<a href="/Record/id">text</a>',
+            $helper('text', '/Record/id')
+        );
+
+        $this->assertEquals(
+            '<a href="#anchor">text</a>',
+            $helper('text', '#anchor')
+        );
+
+        $this->assertEquals(
+            '<a href="#">text</a>',
+            $helper('text', '#')
+        );
+    }
+
+    /**
+     * Test that helper returns unescaped text when href is falsey
+     *
+     * @return void
+     */
+    public function testNoLink()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals('text', $helper('text', ''));
+        // Escapes
+        $this->assertEquals('text&', $helper('text&', null));
+        $this->assertEquals('text<', $helper('text<', false));
+    }
+
+    /**
+     * Test that responds to common inputs
+     *
+     * @return void
+     */
+    public function testAttributes()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals(
+            '<a class="btn" id="login" href="#">text</a>',
+            $helper('text', '#', ['class' => 'btn', 'id' => 'login'])
+        );
+
+        // Skip href
+        $this->assertEquals(
+            '<a href="#" class="btn" id="login">text</a>',
+            $helper('text', ['href' => '#', 'class' => 'btn', 'id' => 'login'])
+        );
+
+        // String
+        $this->assertEquals(
+            '<a class="btn" href="#">text</a>',
+            $helper('text', '#', 'btn')
+        );
+
+        // No href but attributes
+        $this->assertEquals(
+            '<span class="btn">text</span>',
+            $helper('text', null, 'btn')
+        );
+        $this->assertEquals(
+            '<span class="btn">text</span>',
+            $helper('text', ['class' => 'btn'])
+        );
+        $this->assertEquals(
+            '<span class="btn">text</span>',
+            $helper('text', null, ['class' => 'btn'])
+        );
+    }
+
+    /**
+     * Test escaping with other plugins
+     *
+     * @return void
+     */
+    public function testEscaping()
+    {
+        $helper = $this->getHelper();
+
+        $recordLink = $this->getRecordLink();
```

Comment:
Do we really need to create a RecordLink helper in this test? Since it's a unit test for MakeLink, it seems like overkill to use a different "real" object for testing. Can't we just hard-code the outputs that we know RecordLink will give us instead?

Reply:
(Of course, if there are assumptions made here that depend on behaviors that are not already included in RecordLinkTest, we should expand/improve that test... but I think that's a separate issue).

Reply:
It was an escaping test and I wanted to be sure but we could completely exclude these tests if you don't want to pull in RecordLink.

Reply:
My point is that you know what the RecordLink helper is going to return. In the context of this text, you can hard-code that return value as the input to the MakeLink helper instead of creating a real RecordLink helper and running it, because you are not testing the RecordLink helper here. The RecordLink functionality is already being covered by https://github.com/vufind-org/vufind/blob/dev/module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/RecordLinkTest.php (and I'm guessing it's here simply because this test started life as a copy of that test).